src/gsi/apply_scaledepwgts.f90

```diff
@@ -59,53 +59,57 @@ subroutine init_mult_spc_wgts(jcap_in)
   implicit none
 
   integer(i_kind),intent(in   ) :: jcap_in
-  real(r_kind),allocatable      :: totwvlength(:)
 
   integer(i_kind) i,ii,j,k,l,n,kk,nsigend
   integer(i_kind) ig
   real(r_kind) rwv0,rtem1,rtem2
   real (r_kind):: fwgtofwvlen
-  integer(i_kind) :: l_sum_spc_weights
+  real(r_kind) :: totwvlength
+  logical :: l_sum_spc_weights
 
   ! Spectral scale decomposition is differernt between SDL-cross and SDL-nocross
   if( r_ensloccov4scl < tiny_r_kind )then
-     l_sum_spc_weights = 1
+     l_sum_spc_weights = .false.
   else
-     l_sum_spc_weights = 0
+     l_sum_spc_weights = .true.
   end if
 
-  allocate(totwvlength(jcap_in))
+  spc_multwgt(0,1)=one
+  do ig=2,nsclgrp
+    spc_multwgt(0,ig)=zero
+  end do
```

Comment:
Again, Thanks for identifying this "tiny" while important bug introduced in various developing stages.

Reply:
Your welcome.  Like to be useful.